.github/workflows/push.yml

```diff
@@ -0,0 +1,141 @@
+name: CI pipeline
+
+on:
+  push:
+    branches: [ master ]
+
+jobs:
+  test-codepropertygraph:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        lfs: true
+    - name: Clone submodules
+      run: git submodule update --init --recursive
+    - name: Set up JDK 1.8
+      run: sudo apt-get install openjdk-8-jdk
+    - name: Install specific sbt version
+      run: sudo apt-get install -y --allow-downgrades sbt=1.3.12
+    - name: Print sbt version
+      run: sbt --version
+    - name: Run tests
+      run: sbt scalafmtCheck test
+
+  test-fuzzypp-linux:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          lfs: true
+      - name: Clone submodules
+        run: git submodule update --init --recursive
+      - name: Install g++-8
+        run: sudo apt-get install gcc-8 g++-8
+      - name: Run cmake on fuzzypp
+        run: cmake -DCMAKE_CXX_COMPILER=/usr/bin/g++-8 ./fuzzypp
+      - name: Run cmake build
+        run: cmake --build . -- -j4
+      - name: Run fuzzyppcli-tests
+        run: "./fuzzypp/bin/fuzzyppcli-tests"
+      - name: Zip artifact
+        run: zip -j ./fuzzyppcli.zip ./fuzzypp/bin/fuzzyppcli
+      - name: Upload zip artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: unix-zip
+          path: ./fuzzyppcli.zip
+
+  test-fuzzypp-windows:
+    runs-on: windows-latest
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          lfs: true
+      - name: Clone submodules
+        run: git submodule update --init --recursive
+      - name: Run cmake on fuzzypp
+        run: cmake -G "Visual Studio 16 2019" -A "x64" ./fuzzypp
+      - name: Run cmake build
+        run: cmake --build . --config Release
+      - name: Run fuzzyppcli-tests
+        run: "./fuzzypp/bin/Release/fuzzyppcli-tests.exe"
+      - name: Zip artifact
+        run: 7z a -r -tzip ./fuzzyppcli-win.zip ./fuzzypp/bin/Release/fuzzyppcli.exe
+      - name: Upload zip artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: windows-zip
+          path: ./fuzzyppcli-win.zip
+
+  release:
```

Comment:
i ported https://github.com/ShiftLeftSecurity/sbt-ci-release-early to use github-actions, which actually makes things easier. The readme is up to date and there's a few working examples already, e.g. https://github.com/ShiftLeftSecurity/overflowdb/blob/master/.github/workflows/release.yml

I'll try to suggest a workflow config, but since it's for release we'll need to merge to test it i guess.